third_party/2and3/redis/connection.pyi

```diff
@@ -125,20 +131,26 @@ class UnixDomainSocketConnection(Connection):
     def __init__(
         self,
         path=...,
-        db=...,
+        db: int = ...,
+        username=...,
         password=...,
         socket_timeout=...,
         encoding=...,
         encoding_errors=...,
         decode_responses=...,
         retry_on_timeout=...,
         parser_class=...,
-        socket_read_size=...,
+        socket_read_size: int = ...,
+        health_check_interval: int = ...,
+        client_name=...,
     ) -> None: ...
+    def repr_pieces(self) -> List[Tuple[Text, Text]]: ...
+
+def to_bool(value: object) -> bool: ...
 
 class ConnectionPool:
     @classmethod
-    def from_url(cls, url: Text, db: Optional[int] = ..., **kwargs) -> ConnectionPool: ...
+    def from_url(cls, url: Text, db: Optional[int] = ..., decode_components: bool = ..., **kwargs) -> ConnectionPool: ...
```

Comment:
As `decode_components` is keyword-only:

```suggestion
    def from_url(cls, url: Text, db: Optional[int] = ..., *, decode_components: bool = ..., **kwargs) -> ConnectionPool: ...
```

Reply:
Are you sure it's keyword-only?
I get this error in stubtest:
> redis.connection.ConnectionPool.from_url is inconsistent, runtime argument "decode_components" is not keyword-only

https://github.com/andymccurdy/redis-py/blob/3.5.3/redis/connection.py#L950

Reply:
You are right, I misread this.